src/components/Editor/index.js

```diff
@@ -102,10 +102,21 @@ class Editor extends Component {
     if (this.props.onKeyDown) {
       this.props.onKeyDown(evt)
     }
+    
     if (evt.keyCode === 9 && !this.props.ignoreTabKey) { // Tab Key
-      document.execCommand('insertHTML', false, '&#009')
-      evt.preventDefault()
-    } else if (evt.keyCode === 13) { // Enter Key
+      if(!this.tabGuarded){
+        document.execCommand('insertHTML', false, '&#009')
+        evt.preventDefault()
+      }
+    }else if (evt.keyCode && evt.keyCode === 27) {//Esc Key
+      this.tabGuarded=true
+      this.ref.classList.add("tabGuarded")
```

Comment:
Can we just replace the `ref` usage and the `tabGuarded` flag with a flag inside the component state? In this case we don't need to be clever about how to handle the element, and can just use the lifecycle instead :)